server/aesdsocket.c

```diff
@@ -0,0 +1,254 @@
+  memset(&hints, 0, sizeof(hints));
+  //memset(&buffer, 0, bufferlen);
+
+  hints.ai_family = AF_INET; //IPv4
+  hints.ai_socktype = SOCK_STREAM;
+  hints.ai_flags = AI_PASSIVE;
+  int status;
+
+  //int getaddrinfo(const char *node, const char *service, const struct addrinfo *hints, struct addrinfo **res)                 
+  if((status = getaddrinfo(NULL, (PORT), &hints, &serverinfo))!=0)
+  {
+    syslog(LOG_ERR, "Error while registering SIGINT %s\n\r", strerror(errno));
+    exit(1);
+  }
+
+   // creating a socket fd
+   //int socket(int domain, int type, int protocol)
+  if((sockfd = socket(serverinfo->ai_family, serverinfo->ai_socktype, 0))< 0)
+  {
+    printf("socket creation failed \n");
+    syslog(LOG_ERR, "socket creation failed %s\n\r", strerror(errno));
+    exit(1);
+  }
+
+  // Handling the resue of port
+  int yes = 1;
+  if(setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes)) < 0)
+  {
+    printf("setsockopt error\n");
+    syslog(LOG_ERR, "setsockopt %s\n\r", strerror(errno));
+    exit(1);
+  }
+  
+  // bind system call
+  //int bind(int sockfd, struct sockaddr *my_addr, int addrlen)
+  if (bind(sockfd, serverinfo->ai_addr, serverinfo->ai_addrlen) < 0)
+  {
+    printf("Bind error \n");
+    syslog(LOG_ERR, "Bind error: %s\n\r", strerror(errno));
+    exit(1);
+  }
+  
+  //int listen(int sockfd, int backlog)
+  if(listen(sockfd, backlog)== -1)
+  {
+    printf ("Listen error \n");
+    syslog(LOG_ERR, "error on syscall Listen : %s\n\r", strerror(errno));
+    exit(1);
+  }
+  
+  // forking to run the program as daemon
+  pid_t pd;
+  pd=fork();
+
+  if(pd<0)
+    return -1;
+  else if(pd>0)
+  {
+    printf("%d", pd);
+    return 0;
+  }
+
+  if( setsid() <0)
+    return -1;
+
+  signal(SIGCHLD, SIG_IGN);
+  signal(SIGHUP, SIG_IGN);
+
+  umask(0);
+  chdir("/");
+
+  close(STDIN_FILENO);
+  close(STDOUT_FILENO);
+  close(STDERR_FILENO); 
+ 
+  while(1)
+  {
+    struct sockaddr_in client_addr;
+    socklen_t client_len = sizeof(client_addr);
+
+    // Accept an incoming connection
+    client_sockfd = accept(sockfd, (struct sockaddr*)&client_addr, &client_len);
+    if(client_sockfd < 0)
+    {
+      printf("accept error \n");
+      close_connections();
+    }
+    
+    char *client_ip = inet_ntoa(client_addr.sin_addr);
+    printf(" Accepted connection from %s \n",client_ip);
+
+    // read the data
+    while(1)
+    {
+      // int bytes_read = read(client_sockfd, buffer, bufferlen);
+      int bytes_read = recv(client_sockfd, buffer, bufferlen, 0 );
+	    if(bytes_read == -1)
+	    {
+	      printf(" syscall recv error \n");
+	      continue; 
+	    }
+	    if(bytes_read == 0)
+	      continue;
+      // printf("read %d bytes of data \n",bytes_read);
+
+      int fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_APPEND, 0766);
+      if(fd == -1)
+      {
+        syslog(LOG_ERR, "Error writing to file %s\n\r", strerror(errno));
+      }
+
+      int write_bytes = write(fd, buffer, bytes_read);
```

Comment:
@Ganeshkm10 what are you assuming about buffer here?  Do you know there was exactly one packet sent?  What if the packet was 1025 bytes?